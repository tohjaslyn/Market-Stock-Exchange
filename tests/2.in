add trader1 100
add trader2 100
add trader3 100
add trader4 100
add trader5 100
import trader1 ABC 100
buy trader2 ABC 10 7.5
buy trader3 ABC 5 10
buy trader4 ABC 15 5
buy trader5 ABC 20 7.5 
sell trader1 ABC 100 6.5
book buy
book sell
inventory trader2
inventory trader1
exit
