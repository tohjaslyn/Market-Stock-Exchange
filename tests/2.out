$ Success.
$ Success.
$ Success.
$ Success.
$ Success.
$ Trader now has 100.00 units of ABC.
$ No trades could be made, order added to buy book.
$ No trades could be made, order added to buy book.
$ No trades could be made, order added to buy book.
$ No trades could be made, order added to buy book.
$ Product sold in part, trades as follows:
trader1->trader3: 5.00xABC for $10.00.
trader1->trader2: 10.00xABC for $7.50.
trader1->trader5: 20.00xABC for $7.50.
$ 0002: BUY 15.00xABC @ $5.00
$ 0004: SELL 65.00xABC @ $6.50
$ ABC
$ Trader has an empty inventory.
$ Have a nice day.
